ocs_ci/ocs/bucket_utils.py

```diff
@@ -2554,6 +2554,48 @@ def get_object_count_in_bucket(io_pod, bucket_name, prefix="", s3_obj=None):
     return len(output.splitlines())
 
 
+def wait_for_bucket_count_stability(
+    mcg_obj, expected_count=None, max_retries=10, retry_interval=5, stability_count=3
+):
+    """
+    Poll CLI until bucket count is stable or expected count is reached.
+
+    Args:
+        mcg_obj: MCG object instance with cli_list_all_buckets method
+        expected_count (int, optional): Expected number of buckets
+        max_retries (int): Maximum number of retry attempts
+        retry_interval (int): Seconds between retries
+        stability_count (int): Number of consecutive identical counts to consider stable
+
+    Returns:
+        tuple: (final_count, is_expected_reached)
+    """
+    previous_counts = []
+    for attempt in range(max_retries):
+        cli_buckets = list(mcg_obj.cli_list_all_buckets())
```

Comment:
1. [cli_list_all_buckets](https://github.com/red-hat-storage/ocs-ci/blob/master/ocs_ci/ocs/resources/mcg.py/#L263-L269) returns a list of strings, so we don't need the conversion
2. Please fix the docstring of `cli_list_all_buckets` - it states the method is returning a set when it doesn't